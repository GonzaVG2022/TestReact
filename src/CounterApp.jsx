import  PropTypes  from "prop-types";
import { useState } from "react";
export const CounterApp = ({value}) => {
  
  const [counter, setCounter] = useState(value);
  const sumar = () => 
    setCounter( counter + 10 )
  
  const restar = () => 
    setCounter( counter - 10 )
  
  const reset = () => 
    setCounter( value )
  
  return (
    <>
    <h1>CounterApp</h1>
    {/* <h2> { value } </h2> */}
    <h1>{counter}</h1>

    <button 
      onClick={sumar}
    >
      +1
    </button>
    <button 
      onClick={restar}
    >
      -1
    </button>
    <button 
      onClick={reset}
    >
      Reset
    </button>
    
    </>
  )
}
CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
