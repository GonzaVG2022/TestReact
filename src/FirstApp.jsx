import  PropTypes  from "prop-types";



export const FirstApp = ({title, name}) => {
  // console.log(props)
  
  return (
    <>
       
        <h4>{title}</h4>
        <h2>{name}</h2>
       
    
    </>
    
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired, //lo obligo a que lo envie
  name: PropTypes.string.isRequired
}//Propertis con types

// definir las prop por defecto
FirstApp.defaultProps = {
  // title :'No hay titulo',
  name: 'No hay nombre'
}