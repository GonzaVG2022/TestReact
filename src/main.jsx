import React from 'react'
import  ReactDOM from 'react-dom/client'
import './styles.css'//styles globales
import { CounterApp } from './CounterApp'
import { FirstApp } from './FirstApp'


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp
            value= {10}
        />
       {/* <FirstApp
        title='Hola soy Flavio'
       /> */}

    </React.StrictMode>
)