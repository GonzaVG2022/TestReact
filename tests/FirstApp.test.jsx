import { render } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirstApp/>', () => { 
    test('debe de hacer match con el snapshot', () => { 
        const title= 'Hola soy Flavio';
        // render(<FirstApp title={title}/>)
        const {container} = render(<FirstApp title={title}/>)//
        expect(container).toMatchSnapshot();

     })
 })