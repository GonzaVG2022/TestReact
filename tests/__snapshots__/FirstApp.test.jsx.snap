// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h4>
    Hola soy Flavio
  </h4>
  <h2>
    No hay nombre
  </h2>
</div>
`;
